test/IbanNet.Tests/Localisation/GermanLanguageTest.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using IbanNet.Registry;
+using IbanNet.Validation.Results;
+using IbanNet.Validation.Rules;
+using Xunit;
+
+namespace IbanNet.Localisation
+{
+
+    public class GermanLanguageTest
+    {
+        [Fact]
+        public void GermanLanguageTestInvalidLenght()
+        {
+            System.Globalization.CultureInfo before = System.Threading.Thread.CurrentThread.CurrentCulture;
+            try
+            {
+                System.Threading.Thread.CurrentThread.CurrentCulture =
+                    new System.Globalization.CultureInfo("de");
+                
+                IbanNet.Resources.Culture = new CultureInfo("en-us");
+
+                IsValidLengthRule r = new IsValidLengthRule();
+                var result = r.Validate(new ValidationRuleContext("AT1234")
+                {
+                    Country = new IbanCountry("AT")
+                });
+
+                var resultTyped = result as InvalidLengthResult;
+                if (resultTyped != null)
+                {
+                    Assert.Equal("Der IBAN hat eine falsche Länge.", resultTyped.ErrorMessage);
+                }
+            }
+
+            finally
+            {
+                IbanNet.Resources.Culture = null;
+                System.Threading.Thread.CurrentThread.CurrentUICulture = before;
```

Comment:
Restoring the wrong property (`CurrentCulture`):

[L20](https://github.com/skwasjer/IbanNet/pull/28/files#diff-87475bc9cf484ce05297f72718c71e183c0a1323f8e150a414df1428bdc8ae60R20):
```
System.Globalization.CultureInfo before = System.Threading.Thread.CurrentThread.CurrentCulture;
```